File: android_java_hummus/src/main/java/com/github/klee0kai/hummus/arch/mvp/ISimplePresenter.java
package com.github.klee0kai.hummus.arch.mvp;

public interface ISimplePresenter {

    /**
     * get view state by key
     */
    Object getState(String stateId);

    /**
     * safe view state by key
     */
    void saveState(String stateId, Object state);

    /**
     * subscribe view for data update
     */
    void subscribe(IRefreshView view);

    void unsubscribe(IRefreshView view);

}

File: android_kotlin_hummus/src/main/kotlin/com/github/klee0kai/hummus/arch/mvp/ISimplePresenter.kt
package com.github.klee0kai.hummus.arch.mvp

interface ISimplePresenter {
    /**
     * get view state by key
     */
    fun getState(stateId: String?): Any?

    /**
     * safe view state by key
     */
    fun saveState(stateId: String?, state: Any?)

    /**
     * subscribe view for data update
     */
    fun subscribe(view: IRefreshView?)
    fun unsubscribe(view: IRefreshView?)
}